### PR TITLE
Rivtech mainline

### LINKED ISSUE #3082: Craftable rivtech ammo
### Is your feature request related to a problem? Please describe.

Currently:

5x50 is craftable as original ammo (flechette and penetrator, both have 1 plastic chunk as a requirement), available in base BN

20x66 is craftable as nerfed handmade ammo (flechette, buckshot, and slug, these require 0.2 plastic chunks per shot), and only available in the Craftable gun pack mod

8x40 is only craftable as bootleg ammo (only JHP which is utter trash, 0.125 plastic chunks per bullet), and only available in the Craftable gun pack mod

### Describe the solution you'd like

1. Either move all Rivtech ammo to craftable guns or vanilla. I see the logic though - 5x50 has casings while the rest (8x40, 20x66) is caseless, so basic reloading works for this, while advanced caseless handloading is more 'magic' than vanilla. So this can stay as is, with 5x50 reloadable by default, and 20x66 and 8x40 only craftable with the craftable guns mod.

2. Balance 5x50 to use between 0.1 and 0.2 plastic chunks per bullet. Either by using plastic scrap instead of chunks, or by multiplying the products and other requirements. I am more in favour of multiplying, as plastic chunks can't be disassembled into scrap, and that would need to be added for this recipe to not be useless and require scrapping a plastic bag for every bullet.

3. Add craftable caseless/hvp/incendiary 8x40 ammo. Either nerfed handloaded variant or original factory grade ammo to avoid item bloat.

### Additional context

I'm happy to do all the changes and submit a full PR, I just want input on the above, and, as a side-question, should parts of this remain in the modded-only section, is it reasonable to edit effectively someone elses' mod in the CBN repo?

### PR DESCRIPTION
**Pull request by borsek**
#### Summary
Content: Rivtech ammo mainline and changes, followup from #3088 and #3082.

#### Purpose of change

Fixes #3082.
As per the discussion in #3088, Rivtech ammo crafting has been mainlined. I changed the recipes to be a bit less nonsensical (imho) - comments welcome. The recipes are now only obtained in the Rivtech design binder, except for 8x40 JHP, which is shit ammo.

#### Describe the solution

Recipes for Rivtech ammo crafting were moved to the base game from the Craftable Gun Pack mod. 

In all recipes the main thing was the removal of acid in the recipe, which is replaced by adhesive instead. Similarly, incendiary was largely replaced by gunpowder, except in tracer rounds, where gunpowder was added.

5x50 ammo now uses 1 plastic chunk per 8 bullets, instead of 1:1.
8x40 variants are craftable, using standard ammo requirements and always some silver/gold/platinum/tin/bismuth. HVP rounds use a lot of these as they are very good - balancing and sensibility, as these act as the tungsten cores.
20x66 shots are also fully mainline now.

Obsoletion has been added to the mod so that old ammo gets converted. I have renamed all variants to follow the standard as seen in other ammo recipes ("bootleg" replaced by ", handmade" in item names, or "_reloaded" in item IDs).

#### Testing

Went in-game, recipes work. I had recipe errors due to having learned #3088 recipes on my local save beforehand. Otherwise everything seems to convert well enough. Not sure if another line is needed in obsoletion for "8mm_bootleg" as I only got a "8mm_bootleg_jsp" error. I think some of this is already obsoleted somewhere.

#### Additional context

Rivtech gun good, yes? Shoot boolet, yes? Go, go, shoot gun, comrade-survivor!

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Do not simply change `id.` Without obsolation (adding id to obsolate json) it will cause loading errors on existing saves.
I recommend to keep `id`, but just change the `name`.

**Comment by Firestorm01X2** [comment]
> I have renamed all variants to follow the standard as seen in other ammo recipes ("bootleg" replaced by ", handmade" in item names, or "_reloaded" in item IDs).

There is 2 options. 
1)Either add item to obolate and create new one with new `id`.
2) Keep `id`, but just change the `name`.

**Comment by borsek** [comment]
This was a mess - the item 8x40mm bootleg was in the main game jsons, while the recipe was in the mod jsons. The naming was also all over the place. But you're right, instead of adding the obsoletion to the mod, it's now added to mainline.

So in essence, I already obsoleted the stuff - what I did was change the names, and IDs, adding obsoletion to the old names. Now it's all moved to mainline as well.

File: data/json/recipes/ammo/rifle.json [hunk 1]
```diff
@@ -395,6 +395,135 @@
...
+    "type": "recipe",
+    "result": "8mm_jhp_reloaded",
+    "category": "CC_AMMO",
+    "subcategory": "CSC_AMMO_RIFLE",
+    "skill_used": "fabrication",
+    "skills_required": [ "cooking", 2 ],
+    "difficulty": 8,
+    "time": 35000,
+    "batch_time_factors": [ 60, 5 ],
+    "book_learn": [ [ "recipe_caseless", 5 ], [ "recipe_bullets", 9 ], [ "manual_pistol", 10 ], [ "manual_rifle", 10 ] ],
```
Comment (Firestorm01X2) [inline data/json/recipes/ammo/rifle.json:408]: Probably typo.
I guess there is should be only `recipe_caseless` aka `Rivtech design binder`

Comment (borsek) [reply data/json/recipes/ammo/rifle.json:408]: The recipes are now only obtained in the Rivtech design binder, **except for 8x40 JHP**, which is shit ammo.

It's shit ammo, so the recipe is available earlier - 50 dmg 0 AP makes for ammo vs maybe normal zombies. Can remove it if you think it's OP.

Comment (borsek) [reply data/json/recipes/ammo/rifle.json]: Ehh, on second thought, makes more sense to only get it from the binder. Anyway, changed.

**Review by Firestorm01X2 — APPROVED** [review]
Ready

**Pull request merged**